### PR TITLE
refactor: clarify the position code + prepare field position change.

### PR DESCRIPTION
**Pull request by monperrus**
Necessary for #3393 and later for #3386

### CODE REVIEW

File: src/main/java/spoon/reflect/cu/position/DeclarationSourcePosition.java [hunk 1]
```diff
@@ -15,8 +15,18 @@
...
+	/** returns the last char of the last modifier */
 	int getModifierSourceEnd();
 
+	/** returns the end of the default value
+	 * int i = 0, j =1 => returns the comma
+	 */
+	int getDefaultValueEnd();
+
+	/** sets the position of the end of the default value declaration */
+	DeclarationSourcePosition setDefaultValueEnd(int endDefaultValueDeclaration);
```
Comment (tdurieux) [inline src/main/java/spoon/reflect/cu/position/DeclarationSourcePosition.java:30]: Not sure it is a good idea to make the position mutable. I think we paid attention in the past to keep them unmutable.

Comment (monperrus) [reply src/main/java/spoon/reflect/cu/position/DeclarationSourcePosition.java:30]: I agree, immutability is such a good property. I'll change this.

**Comment by monperrus** [comment]
@nharrand ready for review

File: src/main/java/spoon/support/DefaultCoreFactory.java [hunk 2]
```diff
@@ -786,9 +786,9 @@ public DeclarationSourcePosition createDeclarationSourcePosition(CompilationUnit
 	}
 
 	@Override
-	public BodyHolderSourcePosition createBodyHolderSourcePosition(CompilationUnit compilationUnit, int startSource, int end, int modifierStart, int modifierEnd, int declarationStart, int declarationEnd, int bodyStart, int bodyEnd, int[] lineSeparatorPositions) {
+	public BodyHolderSourcePosition createBodyHolderSourcePosition(CompilationUnit compilationUnit, int nameStart, int nameEnd, int modifierStart, int modifierEnd, int declarationStart, int declarationEnd, int bodyStart, int bodyEnd, int[] lineSeparatorPositions) {
```
Comment (nharrand) [inline src/main/java/spoon/support/DefaultCoreFactory.java:789]:
```suggestion
	public BodyHolderSourcePosition createBodyHolderSourcePosition(
			CompilationUnit compilationUnit, 
			int nameStart, int nameEnd, 
			int modifierStart, int modifierEnd, 
			int declarationStart, int declarationEnd, 
			int bodyStart, int bodyEnd, int[] lineSeparatorPositions) {
```

**Comment by nharrand** [comment]
LGTM!

**Comment by monperrus** [comment]
@nharrand ready

**Comment by monperrus** [comment]
@nharrand now all green

**Pull request merged**